docs/platform-external/use-case-oci-upi.md

```diff
@@ -0,0 +1,576 @@
+    "pod-security.kubernetes.io/audit": "privileged"
+    "pod-security.kubernetes.io/warn": "privileged"
+    "security.openshift.io/scc.podSecurityLabelSync": "false"
+    "openshift.io/run-level": "0"
+    "pod-security.kubernetes.io/enforce-version": "v1.24"
+EOF
+```
+
+- Create the OCI cloud-config secret:
+
+```bash
+# Change Me:
+export CLUSTER_NAME=ChangeMe
+export BASE_DOMAIN=ChangeMe
+export OCI_CONFIG_CLUSTER_REGION=us-sanjose-1
+export OCI_CCM_COMPARTMENT_ID=ChangeMe
+export OCI_VCN_ID=ChangeMe
+export OCI_LB_SUBNET_ID=ChangeMe
+
+# Create the secret manifest
+cat <<EOF1 > ${INSTALL_DIR}/manifests/oci-01-ccm-00-secret.yaml
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: oci-cloud-controller-manager
+  namespace: $OCI_CCM_NAMESPACE
+data:
+  cloud-provider.yaml: $(cat <<EOF2 | base64 -w0
+auth:
+  region: $OCI_CONFIG_CLUSTER_REGION
+  useInstancePrincipals: true
+
+compartment: $OCI_CCM_COMPARTMENT_ID
+vcn: $OCI_VCN_ID
+
+loadBalancer:
+  securityListManagementMode: None
+  subnet1: $OCI_LB_SUBNET_ID
+
+# Optional rate limit controls for accessing OCI API
+rateLimiter:
+  rateLimitQPSRead: 20.0
+  rateLimitBucketRead: 5
+  rateLimitQPSWrite: 20.0
+  rateLimitBucketWrite: 5
+EOF2
+)
+EOF1
+```
+
+- Download manifests from Github and save it in the directory `${INSTALL_DIR}/manifests`
+
+```bash
+export RELEASE=v1.25.0
+
+wget https://github.com/oracle/oci-cloud-controller-manager/releases/download/${RELEASE}/oci-cloud-controller-manager-rbac.yaml -O oci-cloud-controller-manager-rbac.yaml
+
+wget  https://github.com/oracle/oci-cloud-controller-manager/releases/download/${RELEASE}/oci-cloud-controller-manager.yaml -O oci-cloud-controller-manager.yaml
+```
+
+- Patch the RBAC file setting the correct namespace
+
+```bash
+./yq ". | select(.kind==\"ServiceAccount\").metadata.namespace=\"$OCI_CCM_NAMESPACE\"" oci-cloud-controller-manager-rbac.yaml > ${INSTALL_DIR}/manifests/oci-01-ccm-01-rbac.yaml
+```
+
+- Patch the DaemonSet setting the namespace, append the tolerations, mount CA, and add env vars for the kube URL for OpenShift
+
+> `yq` is required to patch the manifests
+
+> https://mikefarah.gitbook.io/yq/operators/multiply-merge
+
+```bash
+cat <<EOF > ./patch1_oci-cloud-controller-manager.yaml
+metadata:
+  namespace: $OCI_CCM_NAMESPACE
+spec:
+  template:
+    spec:
+      tolerations:
+        - key: node.kubernetes.io/not-ready
+          operator: Exists
+          effect: NoSchedule
+        - key: CriticalAddonsOnly
+          operator: Exists
+      volumes:
+        - name: trusted-ca
+          configMap:
+            name: ccm-trusted-ca
+            items:
+              - key: ca-bundle.crt
+                path: tls-ca-bundle.pem
+EOF
+
+cat <<EOF > ./patch2_oci-cloud-controller-manager.yaml
+spec:
+  template:
+    spec:
+      containers:
+        - volumeMounts:
+            - name: trusted-ca
+              mountPath: /etc/pki/ca-trust/extracted/pem
+              readOnly: true
+          env:
```

Comment:
it should be temporary, but for consistency, should we align with the template and source `/etc/kubernetes/apiserver-url.env` instead of setting the environment?

Reply:
yeah, I think so. I was thinking about how to do it in the kubernetes object, instead of changing their cmd statement - let's say for example that the bash is not available in the image.